### DISCUSSION TITLE
DocumentWriter.set_metadata

### DISCUSSION THREAD

**Discussion by vb64** [pymupdf/PyMuPDF#2696] [Looking for help] [answered]
I use DocumentWriter to convert html Story to pdf.

How can I set the META properties for document (doc.set_metadata)?

**Comment by JorjMcKie** [comment]
Depeding on whether you are ok with PDF's standard metadata (no own, non-standard information keys), prepare a dictionary with the following keywords (or a subset of them):
```python
m = {'author': 'as desired',
 'creationDate': "D:20161124064334-04'00'",  # you can use fitz.get_pdf_now()
 'creator': 'as desired',
 'keywords': 'comma-separated',
 'modDate': "D:20200301055940-04'00'",
 'producer': 'as desired, usually the software',
 'subject': 'as desired',
 'title': 'as desired',
}
# then do:
doc.set_metadata(m)
```

**Reply by JorjMcKie** [reply]
I f you want to go beyond the standard metadata keys, lookup the documentation here: https://pymupdf.readthedocs.io/en/latest/recipes-low-level-interfaces.html#how-to-extend-pdf-metadata

**Comment by vb64** [comment]
I use DocumentWriter, not Document. Haw I can access to metadata with writer object?

Without writer.close() and then reopen by fitz.open(saved_file)?

**Reply by JorjMcKie** [reply, marked as answer]
Ah, ok.
You can use DocumentWriter to write to a io.BytsIO(). Then open that memory piece again as a fitz.Document: `doc = fitz.open("pdf", fp.getvalue())`, where `fp = io.BytesIO()` and DocumentWriter has written to fp.

**Reply by vb64** [reply]
Ok, thanks.
